enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -0,0 +1,169 @@
+
+Any secret reference which gets set to this field has to be validated. Such a referenced secret will be
+expected to contain a `kubeconfig` key with a valid kubeconfig configuration for the webhook. In 4.4,
+it will only be possible to configure a single element for the `wehbhookTokenAuthenticators` field
+and the referenced kubeconfigs will only be allowed to use the `-data` forms of their fields to configure
+certificate authority, client certificate and key.
+
+This path is compatible with kubernetes as it is possible to directly map the referenced
+secrets' contents to files and specify these in the kube-apiserver configuration.
+
+#### Backend - kube-apiserver-operator
+
+1. observes any changes in `webhookTokenAuthenticators` field of the `authentication/cluster` resource
+  and performs validation of any references in that field
+2. it pushes any webhook authenticator that passes the validation to a secret in `openshift-kube-apiserver`
+  namespace, the name of the secret is derived from its index in the `spec.webhookTokenAuthenticators`
+  field of the `authentications.config/cluster` object such that `name=webhook-authenticator-<index>`
+3. if no webhook authenticators pass validation, the operator keeps the previous state
+4. redeploys the kube-apiserver static pods while specifying the new
+  mounted files in the `webhookTokenAuthenticators` [config field](https://github.com/openshift/api/blob/992bc3a41fe6c4e5bcf27dc8080f9e5c114c33fc/kubecontrolplane/v1/types.go#L146).
+
+#### Backend - Integrated OAuth Server Token Authentication
+
+1. cluster-authentication-operator observes `authentication.config/cluster` `type` field
+2. if set to "IntegratedOAuth", it creates an `openshift-config/webhook-authentication-integrated-oauth`
+  secret and adds it to the `webhookTokenAuthenticators` field of the `authentication.config/cluster`
+  resource
+  - the above secret contains details that point kube-apiserver to the oauth-apiserver for token reviews
+3. during an authentication flow, kube-apiserver sends `tokenreviews.authentication.k8s.io` object to the
+  oauth-apiserver for review
+4. oauth-apiserver sends a response accordingly to Kubernetes requirements, on success it adds the
+  `system:authenticated:oauth` group to the returned details about the groups of the authenticated user
+  - `system:authenticated:oauth` possesses privileges for users to handle projects via the self-provisioner
+    cluster role
+  - similarly, if any integrator would like their users to have project privileges, they need to add this
+    specific group in their token review responses.
+
+### Common Gotchas When Providing Your Own Authentication Webhook
+
+1. `oauth-proxy` relies on the _/.well-known/oauth-authorization-server_ endpoint
+   and if the integrated oauth-server is turned off and the `spec.oauthMetadata` field
+   in `authentications.config/cluster` is not set   properly, things like grafana
+   and prometheus won't be accessible
```

Comment:
this is "only" a documentation issue?

Reply:
both the issues are something that the person trying to add their own authentication webhook needs to count with

Reply:
So documentation should probably include it as a warning, you're right